src/extensions/storage_macro/dbc.rs

```diff
@@ -0,0 +1,44 @@
+use async_recursion::async_recursion;
+
+use rdbc2;
+
+type Error = Box<dyn std::error::Error + Send + Sync + 'static>;
+
+impl crate::application::CoLink {
+    #[async_recursion]
+    async fn _search_and_generate_query_string(&self, address: &str, key_name: &str) -> Result<String, Error> {
+        let split_key_path: Vec<&str> = key_name.split(':').collect();
+        for i in 0..split_key_path.len() {
+            let current_key_path = format!("{}:{}", address, split_key_path[0..i].join(":"));
```

Comment:
so we are matching from shortest? I thought we are matching from longest (so i is from `split_key_path.len()-1`)? Also, can we show this feature in the test?

Reply:
Done with the rev(), still working on the test